apps/prairielearn/src/pages/administratorInstitutions/administratorInstitutions.html.ts

```diff
@@ -89,3 +113,266 @@ export function AdministratorInstitutions({
+                value="${institution.short_name}"
+              />
+            </div>
+            <div class="form-group">
+              <label for="long_name">Long name</label>
+              <input
+                type="text"
+                class="form-control"
+                id="long_name"
+                name="long_name"
+                value="${institution.long_name}"
+              />
+            </div>
+            <div class="form-group">
+              <label for="display_timezone">Timezone</label>
+              <input
+                type="text"
+                class="form-control"
+                id="display_timezone"
+                name="display_timezone"
+                value="${institution.display_timezone}"
+              />
+            </div>
+            <div class="form-group">
+              <label for="uid_regexp">UID regexp</label>
+              <input
+                type="text"
+                class="form-control"
+                id="uid_regexp"
+                name="uid_regexp"
+                value="${institution.uid_regexp}"
+              />
+            <div class="form-group">
+              <label for="authn_providers">Authn providers <p><small>Note: authn providers must be separated by a comma and a space (i.e, "Azure, Google") </small><p></label>
+              <input
+                type="text"
+                class="form-control"
+                id="authn_providers"
+                name="authn_providers"
+                value="${authn_providers.join(', ')}"
+              />
+            </div>
+            <div class="form-group">
+              <label for="course_instance_enrollment_limit">Course instance enrollment limit</label>
+              <input
+                type="number"
+                class="form-control"
+                id="course_instance_enrollment_limit"
+                name="course_instance_enrollment_limit"
+                value="${institution.course_instance_enrollment_limit}"
+              />
+            </div>
+            <div class="form-group">
+              <label for="yearly_enrollment_limit">Yearly enrollment limit</label>
+              <input
+                type="number"
+                class="form-control"
+                id="yearly_enrollment_limit"
+                name="yearly_enrollment_limit"
+                value="${institution.yearly_enrollment_limit}"
+              />
+            </div>
+          </form>
+        </div>
+        <div class="modal-footer">
+          <button
+            type="button"
+            class="btn btn-secondary"
+            data-dismiss="modal"
+          >
+            Close
+          </button>
+          <button type="submit" class="btn btn-primary" form="edit-institution-${institution.id}">
+            Save changes
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+  `;
+}
+
+function addInstitutionModal({ csrf_token }: { csrf_token: string }) {
```

Comment:
I wouldn't have expected you to know this, but in PT, we have a couple nice affordances for this modal:

- We show help text to guide the admin in creating an institution
- We have a dropdown for timezone selection

<img width="789" alt="Screenshot 2023-12-05 at 14 21 34" src="https://github.com/PrairieLearn/PrairieLearn/assets/1476544/241320f1-486a-4888-a168-0760aee232c8">

For the help text, you can probably copy pretty closely from PT. For the timezone dropdown, I'll copy some of the code that helps drive that from PT, which you should hopefully be able to build on. But please let me know if you have any questions!

```ts
// lib/timezones.ts
import { z } from 'zod';

import { loadSqlEquiv, queryValidatedRows } from '@prairielearn/postgres';

const sql = loadSqlEquiv(import.meta.url);

export const TimezoneCodec = z.object({
  name: z.string(),
  utc_offset: z.number(),
});
export type Timezone = z.infer<typeof TimezoneCodec>;

let memoizedAvailableTimezones: Timezone[] | null = null;
let memoizedAvailableTimezonesByName: Map<string, Timezone> | null = null;

async function getAvailableTimezonesFromDB(): Promise<Timezone[]> {
  const availableTimezones = await queryValidatedRows(sql.select_timezones, [], TimezoneCodec);
  return availableTimezones;
}

export async function getAvailableTimezones(): Promise<Timezone[]> {
  if (memoizedAvailableTimezones == null) {
    memoizedAvailableTimezones = await getAvailableTimezonesFromDB();
  }
  return memoizedAvailableTimezones;
}

export async function getAvailableTimezonesByName(): Promise<Map<string, Timezone>> {
  if (memoizedAvailableTimezonesByName == null) {
    const availableTimezones = await getAvailableTimezones();
    memoizedAvailableTimezonesByName = new Map(availableTimezones.map((tz) => [tz.name, tz]));
  }
  return memoizedAvailableTimezonesByName;
}

export async function getTimezoneByName(name: string): Promise<Timezone> {
  const availableTimezonesByName = await getAvailableTimezonesByName();
  const timezone = availableTimezonesByName.get(name);
  if (timezone == null) {
    throw new Error(`Timezone "${name}" not found`);
  }
  return timezone;
}
```

```sql
-- lib/timezones.sql

-- BLOCK select_timezones
SELECT
  name,
  interval_to_milliseconds (utc_offset) AS utc_offset
FROM
  pg_timezone_names
ORDER BY
  utc_offset,
  name;
```

In essence, we use Postgres's list of timezones as the source of truth for what's allowable and for which options we should give to the user.

Reply:
I copied the help text and lifted `lib/timezones` into PL in commit [402fad6](https://github.com/PrairieLearn/PrairieLearn/pull/9045/commits/402fad6c5a45a0e8ff5187613d3563a1bbf7d8f0).